General/ProgrammaticCodeManipulation.md

```diff
@@ -0,0 +1,136 @@
+# Programmatic Code Manipulation (Pharo 11)
+
+In this article, we will present some code snippets that demonstrate how we can manipulate the source code (packages, classes, methods, variables, traits, etc.) programmatically, without using the SystemBrowser. Those techniques can be very useful for reflective operations, code generation, testing, etc.
+
+This article explains only those operations that **modify** the code. We will write a separate article on **querrying code** (looking for method senders, class references, pragma usages, etc.).
+
+## Packages
+
+### Creating a new package
+
+```st
+package := self packageOrganizer ensurePackage: 'PackageName'.
+```
+
+In Playground, you can use any object or class instead of `self`:
+
+```st
+package := Object packageOrganizer ensurePackage: 'PackageName'.
+```
+
+### Removing a package
+
+```st
+package removeFromSystem.
+```
+
+**Be careful!** In Pharo 11 and earlier, this will not properly remove classes from the system. So make sure to remove all classes before you remove the package.
+
+### Creating a package without installing it
+
+_(how to create an instance of a package without installing it into the image? - e.g., for testing)_
```

Comment:
By definition we cannot because currently we consider that code is installed if it is in a package. 

If you want to manipulate packages that are not in the global organizer you can create a new organizer:

```st
PackageOrganizer new
```

(Note that this name exist only in P12+)

Or better, create a new environment that will have a new package organizer (because the environment of an environmentless organizer will still be the global environment)

```st
SystemDictionary new
```

Or to avoid hard references:

```st
self class environment class new
```